engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -474,31 +478,37 @@ private void validateConstraintsForCurrentGroup(ValidationContext<?> validationC
 			}
 
 			PathImpl currentPath = valueContext.getPropertyPath();
-			for ( Class<?> defaultSequenceMember : defaultGroupSequence ) {
-				valueContext.setCurrentGroup( defaultSequenceMember );
-				boolean validationSuccessful = true;
-				for ( MetaConstraint<?> metaConstraint : metaConstraints ) {
-					// HV-466, an interface implemented more than one time in the hierarchy has to be validated only one
-					// time. An interface can define more than one constraint, we have to check the class we are validating.
-					final Class<?> declaringClass = metaConstraint.getLocation().getDeclaringClass();
-					if ( declaringClass.isInterface() ) {
-						Class<?> validatedForClass = validatedInterfaces.get( declaringClass );
-						if ( validatedForClass != null && !validatedForClass.equals( clazz ) ) {
-							continue;
+
+			while ( defaultGroupSequence.hasNext() ) {
```

Comment:
WDYT, should we try and avoid the loop if its the default group? I'd extract the inner contents (the loop over constraints) into a method and straight go there for the single default group in that case. Avoids some iterator allocations, but complicates backporting. Any thoughts, @hferentschik, @emmanuelbernard`